WooCommerce/Classes/ViewRelated/Dashboard/Blaze/BlazeCampaignDashboardView.swift

```diff
@@ -21,21 +22,34 @@ final class BlazeCampaignDashboardViewHostingController: SelfSizingHostingContro
             }
         }
 
+        let coordinator = BlazeCampaignCreationCoordinator(
+            siteID: viewModel.siteID,
+            siteURL: viewModel.siteURL,
+            source: .myStoreSection,
+            navigationController: parentNavigationController,
+            onCampaignCreated: self.handlePostCreation
+        )
+        self.coordinator = coordinator
+
         rootView.createCampaignTapped = { [weak self] in
-            self?.navigateToCampaignCreation(source: .myStoreSection)
+            guard let self = self else {
+                return
+            }
+            coordinator.start()
         }
 
         rootView.startCampaignFromIntroTapped = { [weak self] productID in
-            // Ensures the intro view is dismissed before navigating to the creation flow
-            self?.parentNavigationController?.dismiss(animated: true) {
-                self?.navigateToCampaignCreation(source: .introView, productID: productID)
+            guard let self = self else {
+                return
+            }
+            self.parentNavigationController.dismiss(animated: true) {
+                coordinator.start()
             }
```

Comment:
Please be careful with nested closures, it's best to reference `self` weakly to avoid any retain cycle:

```suggestion
            self?.parentNavigationController.dismiss(animated: true) {
                self?.coordinator.start()
            }
```